### ISSUE TITLE
Change connector definition's prop from list to array

### ISSUE THREAD

**Issue by konekoya** [oharastream/ohara#1369]
The current API response from the `GET /v0/works`

```
{
  ... more props here
  "connectors": [
    {
      "className": "com.island.ohara.connector.ftp.FtpSink",
      "definitions": [
        ... more definitions here
        {
          "reference": "TOPIC",
          "displayName": "Topics",
          "internal": false,
          "documentation": "the topics used by connector",
          "valueType": "LIST",
          "tableKeys": [],
          "orderInGroup": 3,
          "key": "topics",
          "required": true,
          "defaultValue": null,
          "group": "core",
          "editable": true
        }
      ]
    }
  ]
}
```
Note the `valueType` value of this definition is `LIST`. This creates some level of confusion in the frontend since we often associate `List` to a dropdown or select component. Rename this to `ARRAY` would be a better option :)

Since this change will also affect our UI, some necessary todos are:

- [ ] Use `valueType` in conjunction with `reference` to create the form UI @wu87988622 

~Change connector definition's prop value from list to array @chia7712~ (be a part of above PR)

**Comment by chia7712** [comment]
We can't split it to two PR due to broken APIs. It is hard to keep the compatibility so I upload the patch for backend, and @wu87988622 please apply the patch for your PR.
[fix_1369.txt](https://github.com/oharastream/ohara/files/3250457/fix_1369.txt)

**Comment by chia7712** [comment]
@wu87988622  any update?

**Comment by wu87988622** [comment]
> @wu87988622 any update?

Sorry~
Not yet maybe next work.

**Comment by chia7712** [comment]
@wu87988622 We can address this issue in 0.7 :)

**Comment by wu87988622** [comment]
> @wu87988622 We can address this issue in 0.7 :)

OK
thank you!!

**Comment by konekoya** [comment]
Change the fix version label to 0.7, thanks for the help guys!
